src/main/java/bio/terra/service/snapshotbuilder/SnapshotBuilderSettingsDao.java

```diff
@@ -65,18 +81,47 @@ public SnapshotBuilderSettings upsertSnapshotBuilderSettingsByDataset(
             + " VALUES (:dataset_id, cast(:settings as jsonb))"
             + " ON CONFLICT ON CONSTRAINT snapshot_builder_settings_dataset_id_key"
             + " DO UPDATE SET settings = cast(:settings as jsonb)",
-        Map.of(datasetIdField, datasetId, "settings", jsonValue));
-    return getSnapshotBuilderSettingsByDatasetId(datasetId);
+        Map.of(DATASET_ID_FIELD, datasetId, SETTINGS_FIELD, jsonValue));
+    return getByDatasetId(datasetId);
   }
 
   @Transactional(propagation = Propagation.REQUIRED, isolation = Isolation.SERIALIZABLE)
-  public void delete(UUID datasetId) {
+  public SnapshotBuilderSettings upsertBySnapshotId(
+      UUID snapshotId, SnapshotBuilderSettings settings) {
+    String jsonValue;
+    try {
+      jsonValue = objectMapper.writeValueAsString(settings);
+    } catch (JsonProcessingException e) {
+      throw new BadRequestException("Could not write settings to json", e);
+    }
+    jdbcTemplate.update(
+        "INSERT INTO snapshot_builder_settings (snapshot_id, settings)"
+            + " VALUES (:snapshot_id, cast(:settings as jsonb))"
+            + " ON CONFLICT ON CONSTRAINT snapshot_builder_settings_snapshot_id_key"
```

Comment:
> ON CONFLICT ON CONSTRAINT snapshot_builder_settings_snapshot_id_key

For my own understanding, do you know how this line works/how it maps to this key name? It doesn't look like we create a key with this name. Maybe the key is automatically generated.
Clearly it works because the functionality is tested with upsertBySnapshotIdUpdatesWhenExisting()!

Reply:
As near as I can tell, liquibase provides a default primary key name which you can override when setting up a primary key constraint. I couldn't find the docs where it says what the name is, but I did find some pages that said that we shouldn't be relying on the default name. Not sure it's worth changing now? https://whiteclarkegroup.github.io/liquibase-linter/docs/rules/primary-key-name/

Reply:
Interesting. Yeah, I haven't found any other documentation on this. It seems reasonable to keep it as is given the unit test should break if the primary key name changes.

Reply:
This is a good question. I didn't know this either. Thanks for looking into this